File: Ruby/FuzzyPlayer.rb
#encoding: UTF-8

require_relative 'Dice'
require_relative 'Player'

module Irrgarten
	class FuzzyPlayer < Player
		public
		def initialize(other)
			copy(other)
			@health = Dice.health_reward
		end
		
		def move(direction, valid_moves)
     		pref=super(direction, valid_moves)
     		Dice.next_step(pref,valid_moves,@intelligence)
		end

		def attack()
			sum_weapons+Dice.intensity(@strength)
		end
		
		def defensive_energy()
			sum_shields+Dice.intensity(@intelligence)
		end
		
		def to_string()
			"Fuzzy"+super
		end
		
		#public_class_method :new
	end
end

File: Java/src/irrgarten/FuzzyPlayer.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package irrgarten;

import java.util.ArrayList;

/**
 *
 * @author jorge
 */
public class FuzzyPlayer extends Player{
    FuzzyPlayer(Player other){
        super(other);
        this.setHealth(getHealth()+Dice.healthReward());
    }
    
    
    @Override
    public Directions move(Directions direction, ArrayList<Directions> validMoves){
        Directions pref = super.move(direction, validMoves);
        return Dice.nextStep(pref, validMoves, this.getIntelligence());
    }
    
    @Override
    public float attack(){
        return (this.sumWeapons()+Dice.intensity(this.getStrength()));
    }
    
    @Override
    protected float defensiveEnergy(){
        return (this.sumShields()+Dice.intensity(this.getIntelligence()));
    }
    
    @Override
    public String toString(){
        return "Fuzzy"+super.toString();
    }
}